pkg/gadgettracermanager/containers-map/tracer.go

```diff
@@ -16,16 +16,60 @@ package containersmap
 
 import (
 	"bytes"
+	"errors"
 	"fmt"
+	"os"
+	"path/filepath"
 
 	"github.com/cilium/ebpf"
+	"golang.org/x/sys/unix"
+
+	pb "github.com/kinvolk/inspektor-gadget/pkg/gadgettracermanager/api"
+	"github.com/kinvolk/inspektor-gadget/pkg/gadgettracermanager/pubsub"
 )
 
+// #include "../common.h"
+import "C"
+
 const (
-	BPFMapName = "containers"
+	BPFMapName        = "containers"
+	NameMaxLength     = C.NAME_MAX_LENGTH
+	NameMaxCharacters = NameMaxLength - 1
 )
 
-func CreateContainersMap(pinPath string) (*ebpf.Map, error) {
+type Container = C.struct_container
+
+func copyToC(dest *[NameMaxLength]C.char, source string) {
```

Comment:
Is this trully needed? Can't we use `C.memcpy`?

Reply:
C.memcpy does not seem to exist:
```
could not determine kind of name for C.memcpy
```

[This page](https://stackoverflow.com/questions/25327313/cgo-how-to-convert-string-to-c-fixed-char-array) seems to imply there is no easier way.

Reply:
You will need to `#include <string.h>` from C.go to use this (sorry my comment was unclear).

Reply:
Regarding our last discussion, I think you can convert the golang string to C char* with `C.CString(str)`.
But I am fine with your function.

Reply:
I will keep my function: `C.CString(str)` would allocate a new buffer that needs to be released and it would do a second mem copy.